trailblazer/server/api.py

```diff
@@ -166,21 +165,25 @@ def delete(analysis_id):
 
 @blueprint.route("/query-analyses", methods=["POST"])
 def post_query_analyses():
-    """Return list of analyses matching the query terms"""
-    content = request.json
+    """Return list of analyses matching the query terms."""
+    post_request: Response.json = request.json
     query_analyses = store.analyses(
-        case_id=content.get("case_id"),
-        query=content.get("query"),
-        status=content.get("status"),
-        deleted=content.get("deleted"),
-        temp=content.get("temp"),
-        before=parse_datestr(content.get("before")) if content.get("before") else None,
-        is_visible=content.get("visible"),
-        family=content.get("family"),
-        data_analysis=content.get("data_analysis"),
+        before=datetime.strptime(post_request.get("before"), "%Y-%m-%d").date()
+        if post_request.get("before")
+        else None,
+        case_id=post_request.get("case_id"),
+        data_analysis=post_request.get("data_analysis"),
+        deleted=post_request.get("deleted"),
+        family=post_request.get("family"),
+        is_visible=post_request.get("visible"),
+        query=post_request.get("query"),
+        status=post_request.get("status"),
+        temp=post_request.get("temp"),
     )
-    data = [stringify_timestamps(analysis_obj.to_dict()) for analysis_obj in query_analyses]
-    return jsonify(*data), 200
+    raw_analyses: List[Dict[str, str]] = [
+        stringify_timestamps(analysis_obj.to_dict()) for analysis_obj in query_analyses
```

Comment:
```suggestion
        stringify_timestamps(analysis.to_dict()) for analysis in query_analyses
```

Reply:
There is an outer scope warning for this refactoring. Will do it when the server api is refactored